Error when trying to submit to Google Sheets with a GPS coordinate

#### Software and hardware versions 
Collect v1.5.1-15-g3d6ffbb1, Android 6.0.1, Nexus 5

#### Problem description
Error when trying to submit to Google Sheets with a GPS coordinate. The exception is:

```
java.io.FileNotFoundException: /storage/emulated/0/odk/instances/demo-form_2017-04-03_23-39-07/55.5555555 -555.5555555 0.0 21.112: open failed: ENOENT (No such file or directory)
	at libcore.io.IoBridge.open(IoBridge.java:452)
	at java.io.FileInputStream.<init>(FileInputStream.java:76)
	at com.google.api.client.http.FileContent.getInputStream(FileContent.java:71)
	at com.google.api.client.googleapis.media.MediaHttpUploader.resumableUpload(MediaHttpUploader.java:397)
	at com.google.api.client.googleapis.media.MediaHttpUploader.upload(MediaHttpUploader.java:336)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:427)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
	at org.odk.collect.android.tasks.GoogleSheetsAbstractUploader.uploadFileToDrive(GoogleSheetsAbstractUploader.java:667)
	at org.odk.collect.android.tasks.GoogleSheetsAbstractUploader.uploadOneSubmission(GoogleSheetsAbstractUploader.java:324)
	at org.odk.collect.android.tasks.GoogleSheetsAbstractUploader.uploadInstances(GoogleSheetsAbstractUploader.java:130)
	at org.odk.collect.android.activities.GoogleSheetsUploaderActivity$GoogleSheetsInstanceUploaderTask.doInBackground(GoogleSheetsUploaderActivity.java:555)
	at org.odk.collect.android.activities.GoogleSheetsUploaderActivity$GoogleSheetsInstanceUploaderTask.doInBackground(GoogleSheetsUploaderActivity.java:518)
	at android.os.AsyncTask$2.call(AsyncTask.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:234)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
	at libcore.io.Posix.open(Native Method)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:186)
	at libcore.io.IoBridge.open(IoBridge.java:438)
```

#### Steps to reproduce the problem
1. Change protocol to "Google Sheets, Google Drive"
1. Get a form with a GPS coordinate such as the one described [here](https://nafundi.com/blog/posts/serverless-odk-deployments-with-google-apps/)
1. Fill out a GPS coordinate
1. Finalize and submit the form

#### Other information 
It looks like the coordinates are being treated as a filename. This is likely related to #780 so @shobhitagarwal1612 you may want to take a quick look since you know that code well!

--- a/collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java
+++ b/collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java
@@ -832,7 +832,9 @@ public abstract class GoogleSheetsAbstractUploader extends
                     break;
                 case XmlPullParser.TEXT:
                     String answer = parser.getText();
-                    if (answer.contains(".")) {
+                    Pattern pattern = Pattern.compile("^(.*)\\.(.*)$");
+                    Matcher matcher = pattern.matcher(answer);
+                    if (matcher.find() && !matcher.group(2).matches("[0-9]+")) {
                         mediaToUpload.put(getPath(path), answer);
                     } else {
                         answersToUpload.put(getPath(path), answer);

--- a/collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java
+++ b/collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java
@@ -813,13 +813,13 @@ public abstract class GoogleSheetsAbstractUploader extends
 
         parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, false);
         parser.setInput(in, null);
-        readInstanceFeed(parser, answersToUpload, mediaToUpload);
+        readInstanceFeed(parser, answersToUpload, mediaToUpload, instanceFile);
         in.close();
     }
 
     private void readInstanceFeed(XmlPullParser parser,
                                   HashMap<String, String> answersToUpload,
-                                  HashMap<String, String> mediaToUpload)
+                                  HashMap<String, String> mediaToUpload, File instanceFile)
             throws XmlPullParserException, IOException,
             FormException {
         ArrayList<String> path = new ArrayList<String>();
@@ -832,13 +832,16 @@ public abstract class GoogleSheetsAbstractUploader extends
                     break;
                 case XmlPullParser.TEXT:
                     String answer = parser.getText();
-                    Pattern pattern = Pattern.compile("^(.*)\\.(.*)$");
-                    Matcher matcher = pattern.matcher(answer);
-                    if (matcher.find() && !matcher.group(2).matches("[0-9]+")) {
+
+                    String filename = instanceFile.getParentFile() + "/" + answer;
+                    File file = new File(filename);
+
+                    if (file.isFile()) {
                         mediaToUpload.put(getPath(path), answer);
                     } else {
                         answersToUpload.put(getPath(path), answer);
                     }
+
                     break;
                 case XmlPullParser.END_TAG:
                     path.remove(path.size() - 1);

--- a/collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java
+++ b/collect_app/src/main/java/org/odk/collect/android/tasks/GoogleSheetsAbstractUploader.java
@@ -813,15 +813,16 @@ public abstract class GoogleSheetsAbstractUploader extends
 
         parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, false);
         parser.setInput(in, null);
-        readInstanceFeed(parser, answersToUpload, mediaToUpload, instanceFile);
+        readInstanceFeed(parser, answersToUpload, mediaToUpload, instanceFile.getParentFile());
         in.close();
     }
 
     private void readInstanceFeed(XmlPullParser parser,
                                   HashMap<String, String> answersToUpload,
-                                  HashMap<String, String> mediaToUpload, File instanceFile)
+                                  HashMap<String, String> mediaToUpload, File instanceFolder)
             throws XmlPullParserException, IOException,
             FormException {
+
         ArrayList<String> path = new ArrayList<String>();
 
         int event = parser.next();
@@ -833,7 +834,7 @@ public abstract class GoogleSheetsAbstractUploader extends
                 case XmlPullParser.TEXT:
                     String answer = parser.getText();
 
-                    String filename = instanceFile.getParentFile() + "/" + answer;
+                    String filename = instanceFolder + "/" + answer;
                     File file = new File(filename);
 
                     if (file.isFile()) {

